pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -1123,6 +1128,27 @@ def visit_call(self, node: nodes.Call) -> None:
         self._check_use_list_literal(node)
         self._check_use_dict_literal(node)
 
+    @utils.only_required_for_messages("use-yield-from")
+    def visit_yield(self, node: nodes.Yield) -> None:
+        if not isinstance(node.value, nodes.Name):
+            return
+
+        parent = node.parent.parent
+        if (
+            not isinstance(parent, nodes.For)
+            or isinstance(parent, nodes.AsyncFor)
+            or len(parent.body) != 1
+        ):
+            return
+
+        if parent.target.name != node.value.name:
+            return
+
+        if isinstance(node.frame(), nodes.AsyncFunctionDef):
+            return
+
+        self.add_message("use-yield-from", node.lineno, node, confidence=HIGH)
```

Comment:
I guess the for node makes more sense here (so the error appear on the whole for node instead of the yield).
```suggestion
        # Not sure about this one, need checking
        self.add_message("use-yield-from", node.lineno, parent, confidence=HIGH)
```